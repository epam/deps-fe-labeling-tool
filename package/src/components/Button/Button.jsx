import React from 'react'
import { StyledButton } from './Button.styles'

const ButtonType = {
  DEFAULT: 'default',
  LINK: 'link',
  PRIMARY: 'primary'
}

const Button = React.forwardRef(
  (props, ref) =>
    <StyledButton
      ref={ref}
      {...props}
    />
)

Button.displayName = 'Button'

export {
  Button,
  ButtonType
}
