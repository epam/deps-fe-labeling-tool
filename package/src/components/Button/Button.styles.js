import { Button as AntdButton } from 'antd'
import 'antd/lib/button/style/index.less'
import styled from 'styled-components'

const StyledButton = styled(AntdButton)`
  padding: 0;
`

export {
  StyledButton
}
