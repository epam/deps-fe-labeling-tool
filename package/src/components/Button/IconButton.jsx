import React from 'react'
import PropTypes from 'prop-types'
import { childrenShape } from '@/utils/propTypes'
import { IconButton as Button } from './IconButton.styles'

const IconButton = React.forwardRef(
  ({ icon, onClick, className, disabled, ...restProps }, ref) =>
    <Button
      className={className}
      onClick={onClick}
      disabled={disabled}
      ref={ref}
      {...restProps}
    >
      {icon}
    </Button>
)

IconButton.propTypes = {
  icon: childrenShape.isRequired,
  onClick: PropTypes.func,
  className: PropTypes.string,
  disabled: PropTypes.bool
}

IconButton.displayName = 'IconButton'

export {
  IconButton
}
