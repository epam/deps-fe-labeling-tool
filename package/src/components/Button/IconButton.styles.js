import { Button as AntdButton } from 'antd'
import 'antd/lib/button/style/index.less'
import styled from 'styled-components'
import { COLORS } from '@/theme/theme.default'

const IconButton = styled(AntdButton)`
  margin: 0;
  padding: 0;
  border: 1px solid ${COLORS.PRIMARY_3};
  box-shadow: none;

  &:hover {
    background-color: ${COLORS.PRIMARY_3};
    box-shadow: 0 0 3px ${COLORS.SHADOW_5};
  }

  &:focus, &:active {
    outline: none;
    background-color: ${COLORS.PRIMARY_3};
    border: 1px solid ${COLORS.PRIMARY_3};
  }
`

export {
  IconButton
}
