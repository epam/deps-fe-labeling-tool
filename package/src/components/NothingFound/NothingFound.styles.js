import styled from 'styled-components'

const ContentWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding-top: 3rem;
`

const Title = styled.span`
  width: fit-content;
`

export {
  ContentWrapper,
  Title
}
