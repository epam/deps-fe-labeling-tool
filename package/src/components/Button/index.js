import { Button, ButtonType } from './Button'
import { IconButton } from './IconButton'
import { TextButton } from './TextButton.styles'

Button.Icon = IconButton

export {
  Button,
  ButtonType,
  TextButton
}
