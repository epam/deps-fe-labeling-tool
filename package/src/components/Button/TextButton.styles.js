
import styled from 'styled-components'
import { Button } from '@/components/Button'

const TextButton = styled(Button)`
  display: flex;
  align-items: center;
  border: none;
  background-color: transparent;
  box-shadow: none;
  text-align: left;

  &:hover,
  &:active,
  &:focus {
    border: none;
    box-shadow: none;
    background: transparent;
  }

  &:after {
    display: none;
  }
`

export {
  TextButton
}
