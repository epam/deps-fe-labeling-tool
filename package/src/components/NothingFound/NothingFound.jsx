
import React from 'react'
import PropTypes from 'prop-types'
import { Button, ButtonType } from '@/components/Button'
import { ContentWrapper, Title } from './NothingFound.styles'

const RESOURCE_NO_MATCH_FOUND = 'No match found'
const RESOURCE_SHOW_ALL = 'Show all'

const NothingFound = ({
  resetFilter
}) => (
  <ContentWrapper>
    <Title>{RESOURCE_NO_MATCH_FOUND}</Title>
    <Button
      type={ButtonType.LINK}
      onClick={resetFilter}
    >
      {RESOURCE_SHOW_ALL}
    </Button>
  </ContentWrapper>
)

NothingFound.propTypes = {
  resetFilter: PropTypes.func.isRequired
}

export {
  NothingFound
}
