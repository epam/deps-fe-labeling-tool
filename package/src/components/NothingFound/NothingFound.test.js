
import React from 'react'
import { shallow } from 'enzyme'
import { Button } from '@/components/Button'
import { NothingFound } from './NothingFound'

describe('Container: NothingFound', () => {
  let wrapper
  let defaultProps
  beforeEach(() => {
    defaultProps = {
      resetFilter: jest.fn()
    }

    wrapper = shallow(<NothingFound {...defaultProps} />)
  })

  it('should render correct layout', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call resetFilter when calling to button onClick', () => {
    wrapper.find(Button).props().onClick()

    expect(defaultProps.resetFilter).toHaveBeenCalled()
  })
})
