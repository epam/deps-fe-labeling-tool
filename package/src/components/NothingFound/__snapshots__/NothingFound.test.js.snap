// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container: NothingFound should render correct layout 1`] = `
<NothingFoundstyles__ContentWrapper>
  <NothingFoundstyles__Title>
    No match found
  </NothingFoundstyles__Title>
  <Button
    onClick={[MockFunction]}
    type="link"
  >
    Show all
  </Button>
</NothingFoundstyles__ContentWrapper>
`;
